core/src/main/java/org/gbif/pipelines/core/functions/interpretation/DayInterpreter.java

```diff
@@ -0,0 +1,58 @@
+package org.gbif.pipelines.core.functions.interpretation;
+
+import org.gbif.pipelines.core.functions.interpretation.error.Issue;
+import org.gbif.pipelines.core.functions.interpretation.error.IssueType;
+import org.gbif.pipelines.core.functions.interpretation.error.Lineage;
+import org.gbif.pipelines.core.functions.interpretation.error.LineageType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * interprets day and adds issues and lineages with it
+ */
+public class DayInterpreter implements Interpretable<String, Integer> {
+
+  private static final int min_day = 1;
+  private static final int max_day = 31;
+
+  @Override
+  public Integer interpret(String input) throws InterpretationException {
+    Integer x;
+    List<Issue> issues = new ArrayList<>();
+    List<Lineage> lineages = new ArrayList<>();
+
+    try {
+      x = Integer.parseInt(input);
```

Comment:
there's a GBIF NumberParser that can be re-used instead of implementing parsing logic form sratch

Reply:
I avoided it because it did not throw any exception if parse failed, instead it handles it giving null which hides the difference between data was null or it was set to null. Generally ParseResult class gives an error of throwable type.